tests/publiccloud/prepare_instance.pm

```diff
@@ -37,7 +37,7 @@ sub run {
     my $provider = $self->provider_factory();
     my %instance_args;
     $instance_args{check_connectivity} = 1;
-    $instance_args{ignore_wrong_pubkey} //= 0;    # parameter set for wait_for_ssh: 0 = stop on publickey error.
+    $instance_args{ignore_wrong_pubkey} //= 1; # parameter set for wait_for_ssh: 0 = stop on publickey error; 1 = retry to connect until other events or timeout.
```

Comment:
except one exception where on instance reboot there is chroot happening we not expecting public key change and before it never was an issue do you know why now we should accept it ?

Reply:
In the poo note-11 I explained what the failed tests logs are showing.

Let we take the failed test https://openqa.suse.de/tests/11401949.

After ssh port ok, The [ssh keyscan](https://openqa.suse.de/tests/11401949#step/prepare_instance/75) provides 3 keys: ed25519, rsa, nistp256.
In the next [ssh run](https://openqa.suse.de/tests/11401949#step/prepare_instance/83) we have as result a "strange temporay" error result `kex_exchange_identification: read: Connection reset by peer`; _strange_ because this error has been found _always present_ even in _passed_ tests, _temporary_ because it does not affect the next key resolution, see in log `Reloading OpenSSH Daemon` then `Accepted publickey`.

In the [journal]( https://openqa.suse.de/tests/11401949/file/prepare_instance-journalctl.log), we see the key checks sequence provided by the instance:
```
...
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 sshd[1614]: error: kex_exchange_identification: Connection closed by remote host
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 sshd[1615]: Connection closed by 195.135.221.27 port 46952 [preauth]
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 sshd[1617]: Connection closed by 195.135.221.27 port 46926 [preauth]
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 sshd[1619]: Connection closed by 195.135.221.27 port 46942 [preauth]
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 sshd[1616]: Unable to negotiate with 195.135.221.27 port 46960: no matching host key type found. Their offer: sk-ecdsa-sha2-nistp256@openssh.com [preauth]
Jun 21 00:40:01 opetnqa-suse-de-de92d346dfd3bcf6 sshd[1618]: Unable to negotiate with 195.135.221.27 port 46962: no matching host key type found. Their offer: sk-ssh-ed25519@openssh.com [preauth]
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 systemd[1]: Startup finished in 5.261s (kernel) + 3.124s (initrd) + 20.611s (userspace) = 28.997s.
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 systemd[1]: Reloading OpenSSH Daemon.
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 sshd[1583]: Received SIGHUP; restarting.
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 sshd[1583]: Server listening on 0.0.0.0 port 22.
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 sshd[1583]: Server listening on :: port 22.
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 systemd[1]: Reloaded OpenSSH Daemon.
Jun 21 00:40:01 openqa-suse-de-de92d346dfd3bcf6 sshd[1630]: Accepted publickey for susetest from 195.135.221.27 port 46984 ssh2: RSA SHA256:MqNeSY1qN7676L1LFG+VYw/LcQRLZAFHlEtOskViQHM
...
```
The Perl test driver `waith_for_ssh` in this case is set to die at first error, therefore the first `Unable to negotiate...` leads to fail.

Reply:
Isn't that "strange" error just the master connection being killed because of the server has been restarted?

Reply:
Maybe something like this could help?
```bash
ssh -O exit sut
```

Reply:
It is in the create instance phase, the first access to the new instance, so isn't a case of server restart

Reply:
but anyway I saw thet error present in every P.C. test journal once, passed or failed

Reply:
Also a more thing to note is that line `... systemd[1]: Startup finished in...` that is quite the result of the `systemd-analyze` command: it means that the instance is up and running.

Reply:
@m-dati Please see https://progress.opensuse.org/issues/130444#note-12 - I think this is just a simple race condition, and retrying instead of ignoring would be the better approach.

Reply:
Yes, thanks in fact this PR means to retry when that errors occur and [re-loop to next checks](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/cfc785052e30e616046851ac0a7ce57c9a4dce16/lib/publiccloud/instance.pm#L388).

Reply:
In that case, I think this is a valid solution.

For my future self: Please note that `ignore_wrong_pubkey` does not mean that the ssh authentication is weakened, it just means that the call does not fail immediately. It effectively means to retry on failures related to ssh public key errors. We do not ignore them, but re-try.

Reply:
OK,thanks.
Anyway, adding more complexity, but could it be useful to upload a more log, where to store all resulted failure lines (pubkey ,...), before to go next retry and notify at end in the report info something like: 'other errors retries, see log xxx' ?
Otherwise no log, but a simpler _other_errors_retries_  counter :-) ?